docs/load_balancer.md

```diff
@@ -0,0 +1,278 @@
+# Load balancing & publicly accessible services
+
+This document describes how kubernetes services are made accessible outside the cluster.
+
+There are two main mechanisms:
+
+* The service can be exposed through a load-balancer managed by k8s.
+* k8s can also expose services without an associated load-balancer; the service could be reached via DNS SRV records,
+or by a load-balancer configured by a helper process, or even by a client using the k8s API directly for discovery.
+_(new functionality)_
+
+## Public ports
+
+To be exposed outside the k8s cluster, the service must be on known ports on externally reachable
+IPs.  For version 1 public services, k8s listens on _every_ external IP address on every minion, and assigns
```

Comment:
nit: "every external IP is not quite right because external IP is an intrinsically cloud-provider specific thing.  Instead, I would say something like "k8s exposes each public service port on a common port on each node."